elixir/apps/domain/lib/domain/directories/okta/sync_state/changeset.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Domain.Directories.Okta.SyncState.Changeset do
+  use Domain, :changeset
+  alias Domain.Directories.Okta.SyncState
+
+  @fields ~w[
+    full_user_sync_started_at
+    full_user_sync_finished_at
+    full_group_sync_started_at
+    full_group_sync_finished_at
+    full_member_sync_started_at
+    full_member_sync_finished_at
+    delta_user_sync_started_at
+    delta_user_sync_finished_at
+    delta_group_sync_started_at
+    delta_group_sync_finished_at
+    delta_member_sync_started_at
+    delta_member_sync_finished_at
+  ]a
+
+  def changeset(%SyncState{} = sync_state, attrs) do
```

Comment:
I know this is still in draft so you may already have this in mind, but was just thinking it might be nice to have some validations on this changeset function to make sure things like `*_finished_at` times aren't before `*_started_at` times

Reply:
I gave that a little thought and I don't know how strict we want to be here. Consider:

- Jobs can fail in which case started_at > finished_at (you could argue we should explicitly "fail" the job I suppose)
- These will all be nil upon a newly-created provider

I'll give it more thought but will probably save these validations for later.

Reply:
Yeah, that's definitely a good edge case to think through.  My initial thought there would be when setting a `started_at` you always clear out the `finished_at` for that type.  Or like you said, if the job fails we just set the `finished_at` time at that point.

Not a critical thing, so understand if you want to think on it

Reply:
Updated the changeset with validations around this so it should be rock-solid.